foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/expressions/Expression.java

```diff
@@ -1981,6 +1981,11 @@ public Expression sql(String sql, List arguments) {
         int start = 0;
         int index = sql.indexOf('?');
         while (index != -1) {
+            // ? can be escaped as ?? to support ? as a native SQL operator (e.g. on Postgres)
+            if (index < sql.length() - 1 && sql.charAt(index + 1) == '?') {
+                index = sql.indexOf('?', index + 2);
+                continue;
+            }
```

Comment:
It's about line behind 1996 with `anOperator.printsAs(v);`
please add there

```
// replace ?? with ? in code which is passed as a part of SQL into DB to avoid issue like
//Error Message = ORA-40597: JSON path expression syntax error ('$??(@.id == 1007)')
//JZN-00228: Question mark must be followed by parenthetical expression (line 1, position 3)
for (int i = 0; i < anOperator.getDatabaseStrings().length; i++) {
    anOperator.getDatabaseStrings()[i] = anOperator.getDatabaseStrings()[i].replace("??", "?");
}
```

Why? Oracle DB doesn't accept `??` sequence in SQL (Json) queries.
`anOperator.printsAs(v);` convert Expression into Strings which are passed/printed into DB SQL query.
Please let me know if this change suits with Postgres.

Reply:
I see. Makes sense for Oracle, but this doesn't work for the Postgres JDBC driver, which expects '??' in the prepared statement in order to not confuse the '?' [in the statement with positional parameters.](https://jdbc.postgresql.org/documentation/query/#using-the-statement-or-preparedstatement-interface) Somehow the Oracle JDBC drives knows the difference. Do we need a change in some writer elsewhere that again replaces '?' with '??' for Postgres only? I don't know the code enough to know where to do that.

Reply:
I gave it an attempt. See below.